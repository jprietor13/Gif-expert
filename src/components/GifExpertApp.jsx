import React, { useState } from 'react'

const GifExpertApp = () => {
  const categories = ["Bleach", "Dragon Ball Z", "Naruto"];
  const [categorie, setCategorie] = useState(categories);

  const handleAddCategorie = () => {
    const newCategorie = "Samurai X";
    setCategorie([...categories, newCategorie])
  }
  return (
    <div>
      <h2>GifExpertApp</h2>
      <hr />
      <button type="button" onClick={handleAddCategorie}>Agregar</button>
      <ol>
        {categorie.map(category =>
          <li key={category}>{category}</li>
        )}
      </ol>
    </div>
  )
}

export default GifExpertApp;
