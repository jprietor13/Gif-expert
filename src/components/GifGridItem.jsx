import React from 'react';

export const GifGridItem = ({ images }) => {
  return (
    <div className="card animate__backInDown">
      <img src={images.url} alt={images.title} />
      <p>{images.title}</p>
    </div>
  )
}

