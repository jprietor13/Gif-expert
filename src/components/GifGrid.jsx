import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifData } from '../helpers/getGifData';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifData(category)
      .then(images => setImages(images))
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="grid-card">
        {images.map(item =>
          <GifGridItem key={item.id} images={item} />
        )}
      </div>
    </>
  )
}

