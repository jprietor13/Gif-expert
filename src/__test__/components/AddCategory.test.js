import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe("Pruebas a <AddCategory />", () => {

  const setCategorie = () => { };
  const wrapper = shallow(<AddCategory categories={setCategorie} />);
  test("Debe validar el correcto funcionamiento del componente", () => {
    expect(wrapper).toMatchSnapshot();
  })
})