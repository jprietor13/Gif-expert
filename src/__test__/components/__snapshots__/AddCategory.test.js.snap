// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <AddCategory /> Debe validar el correcto funcionamiento del componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar..."
    type="text"
    value=""
  />
</form>
`;
